src/Cli/dotnet/Commands/Run/VirtualProjectBuildingCommand.cs

```diff
@@ -211,7 +205,7 @@ public override int Execute()
             {
                 var buildRequest = new BuildRequestData(
                     CreateProjectInstance(projectCollection),
-                    targetsToBuild: [BuildTarget]);
+                    targetsToBuild: MSBuildArgs.RequestedTargets ?? ["Build"]);
```

Comment:
Does this mean that `dotnet build file.cs -t:MyTarget` now works? Consider capturing that in a test.

Reply:
Great idea to add a test - turns out there was a bug in the Target-arg parsing on re-parsing (like we do).